### PR TITLE
Fix iPad share

### PR DESCRIPTION
**Pull request by EricBAndrews**
# Checklist
- [x] I have read [CONTRIBUTING.md](./CONTRIBUTING.md)
- [x] I have described what this PR contains
- [x] This PR addresses one or more open issues that were assigned to me:
      - #578 
- [x] If this PR alters the UI, I have attached pictures/videos

# Pull Request Information

This PR addresses an issue where sharing from iPad would crash the app. This was due to the use of `showShareSheet`, which was not iPad-compatible. This PR replaces all instance of `showShareSheet` with a `ShareLink`.

To support this, it also performs a refactor of how menu items work:
- `MenuFunction` is now an enum with an associated type, either `ShareMenuFunction` or `StandardMenuFunction`.
- `MenuButton` now displays a `ShareLink` for `MenuFunction.share` and a regular menu button for `MenuFunction.standard`.
- Destructive actions now have their own `ViewModifier` for easy reuse, and have been populated everywhere they're needed

### CODE REVIEW

File: Mlem/Extensions/Encodable+Export.swift
Comment (boscojwho) [inline Mlem/Extensions/Encodable+Export.swift:1]: Forgot to replace this?

File: Mlem/Models/User-Interactable/Menu Function.swift [hunk 1]
```diff
@@ -7,8 +7,52 @@
...
+    }
+    
+    static func shareMenuFunction(url: URL) -> MenuFunction {
+        MenuFunction.share(ShareMenuFunction(url: url))
+    }
+}
+
+/// MenuFunction to open a ShareLink
+struct ShareMenuFunction: Identifiable {
+    var id: String { url.description }
```
Comment (boscojwho) [inline Mlem/Models/User-Interactable/Menu Function.swift:49]: Use `url.absoluteString` instead?

File: Mlem/Views/Shared/Comments/Comment Item Logic.swift [hunk 2]
```diff
@@ -226,18 +226,11 @@ extension CommentItem {
...
         if let url = URL(string: hierarchicalComment.commentView.comment.apId) {
-            ret.append(MenuFunction(
-                text: "Share",
-                imageName: "square.and.arrow.up",
-                destructiveActionPrompt: nil,
-                enabled: true
-            ) {
-                showShareSheet(URLtoShare: url)
-            })
+            ret.append(MenuFunction.shareMenuFunction(url: url))
```
Comment (JakeShirley) [inline Mlem/Views/Shared/Comments/Comment Item Logic.swift:229]: This is really nice cleanup.

File: Mlem/Extensions/View Modifiers/View+DestructiveConfirmation.swift [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+                    Text(destructivePrompt)
+                }
+            }
+    }
+}
+
+extension View {
+    /// View modifier to attach a destructive action confirmation.
+    ///
+    /// To use, add the following to the view in which you use it:
```
Comment (JakeShirley) [inline Mlem/Extensions/View Modifiers/View+DestructiveConfirmation.swift:36]: Just curious but will the compiler tell you that you are missing these things if you use this view modifier without adding these members?

**Review by JakeShirley — APPROVED** [review]

**Pull request merged**